### DISCUSSION TITLE
Waking from sleep

### DISCUSSION THREAD

**Discussion by Inclsevd** [o0Zz/sys-con#23] [Q&A]
I don't know if it's a bug or expected but I noticed it's possible to wake the console from sleep with the home button with a controller connected via usb and sys-con, however it won't work very reliably and I need to spam the home button on the controller until finally it wakes. Any insight on that?

Thanks

**Comment by texita** [comment]
Not sys-con, i got the same behavior with sys-con disabled

**Comment by Inclsevd** [comment]
What was your testing procedure? If I connect an official Switch Pro Controller and turn on wired communication and connect via usb I can wake the console reliably, although it's possible that it still wakes via bluetooth. I also had pairing problems when removing it from the bluetooth database so I couldn't test properly.

**Comment by texita** [comment]
- disabled sys-con
- connected a pro controller (third party are not working without sys-con)
- put switch in sleep mode (used both: button on switch, holding home on controller and set sleep)
- wait a minute and try to wake up with home button

As i said, i got same behavior as you that are using sys-con enabled. Sometimes wake up instantly, sometimes not.
You are also mentioning bluetooth, sys-con hasn't got any bluetooh feature for the controllers, for that you need to ask in MissionControl support

**Comment by Inclsevd** [comment]
That's not a proper confirmation that official usb wake has the same issue and you didn't read my message properly either.

1. You didn't mention anything about enabling Pro Controller wired communication in settings.
2. I didn't say anything about sys-con having bluetooth support so I don't know why you bring that up.

If you read my message again, I mentioned bluetooth because it's possible that if the Pro Controller is paired it still uses bluetooth to wake even with wired communcation enabled.

**Comment by texita** [comment]
Yes, i had pro controller wired communication in settings enabled. I tried 3 pro controllers (hori battle pad, pdp wired controller, powerA wired controller) so all of them without bluetooh support.

i'm keep saying that this has not to do with sys-con because you started this discussion in sys-con Q&A with this

> I don't know if it's a bug or expected

So, it's not a bug because has nothing to do with sys-con this behavior

**Comment by Inclsevd** [comment]
You still don't understand. So for the third time:

We do not have confirmation that this issue occurs with an official Pro Controller in usb mode because it's possible that it still uses bluetooth to perform the wake, even with wired communcation enabled.

edit: I just re-read and I see you now mentioned some wired only controllers when before you only mentioned the Pro Controller.

Also yes I started it specifically in _discussion_ because I wasn't sure if it was a bug or an issue with sys-con or something that happens even on official firmware. I was looking for some insight from @o0Zz who might know why this happens from his work on the usb protocol.

**Comment by o0Zz** [comment]
I'm not an expert on all Nintendo Switch behaviours ;), but as far as I can tell from the code, nothing special is done for this.

If it can provide more information, today's sys-con automatically releases the controller when the console goes into sleep mode:
![image](https://github.com/o0Zz/sys-con/assets/3352109/13878c8e-d60e-45f8-a0d0-76f344d599b7)

This behaviour is inherited from the original version of sys-con. We can try to tweak this part of the code and see if something different happens. 
On my side, I tried plug in my XBOX360 controller (wired only) - it has a led indicator and allows me to understand a bit more what the switch does (when USB is power on/off and when sys-con takes control) Here are my observations:

- When the switch goes into sleep mode, the controller turns off.
- Then maybe 5s later the controller will turn on automatically (I can see the led player 1 is turn on meaning sys-con take the control here and setup the controller).
- At this stage if I press the home button it works reliably - if I try to press the home button before that, nothing happens.

I have no idea if this is what you have observed too.

I'm not sure what does the switch exactly here, why the switch power-off the controller, then power-on it again later, but for sure this power-on/power-off is not a sys-con behavior but more a switch behavior.

EDIT: Another observation, if we try to plug in the controller (Via USB) while the switch is in sleep mode, the controller will never turn on. This means that the switch only keeps the USB port PowerOn if something is plugged into it before it goes into sleep mode.
